src/common/thread_function.h

```diff
@@ -0,0 +1,71 @@
+/*

```

Comment:
Thread, clock and memory allocaiton functions can preferably moved to a common set of platform functions reused across all tests.

Reply:
I will aligned the platform method with this commit https://github.com/FreeRTOS/Labs-FreeRTOS-Libraries-Integration-Tests/commit/1026199282a9cbdb39ce498a11871112f0b20671.

Reply:
Add and use the following platform function instead of using the thread_function.h
```
/**
 * @brief Thread handle data structure definition.
 */
typedef void * FRTestThreadHandle_t;

/**
 * @brief Thread function to be executed in ThreadCreate_t function.
 *
 * @param[in] pParam The pParam parameter passed in ThreadCreate_t function.
 */
typedef void ( * FRTestThreadFunction_t )( void * pParam );

/**
 * @brief Delay function to wait for at least specified amount of time.
 *
 * @param[in] delayMs Delay in milliseconds.
 */
void FRTest_TimeDelay( uint32_t delayMs );

/**
 * @brief Thread create function for test application.
 *
 * @param[in] threadFunc The thread function to be executed in the created thread.
 * @param[in] pParam The pParam parameter passed to the thread function pParam parameter.
 *
 * @return NULL if create thread failed. Otherwise, return the handle of the created thread.
 */
FRTestThreadHandle_t FRTest_ThreadCreate( FRTestThreadFunction_t threadFunc,
                                          void * pParam );

/**
 * @brief Timed thread join function to wait for the created thread exit.
 *
 * @param[in] threadHandle The handle of the created thread to be waited.
 * @param[in] timeoutMs The timeout value of to wait for the created thread exit.
 *
 * @return 0 if the thread exits within timeoutMs. Other value will be regarded as error.
 */
int FRTest_ThreadTimedJoin( FRTestThreadHandle_t threadHandle,
                            uint32_t timeoutMs );

/**
 * @brief Malloc function to allocate memory for test.
 *
 * @param[in] size Size in bytes.
 */
void * FRTest_MemoryAlloc( size_t size );

/**
 * @brief Free function to free memory allocated by FRTest_MemoryAlloc function.
 *
 * @param[in] size Size in bytes.
 */
void FRTest_MemoryFree( void * ptr );
```

Thread_function.h and memory_function.h will be removed in another in case other test still using these functions.